packages/perseus/src/widgets/interactive-graphs/locked-line.tsx

```diff
@@ -15,52 +16,72 @@ type Props = LockedLineType & {
 };
 
 const LockedLine = (props: Props) => {
-    const {
-        color,
-        lineStyle,
-        kind,
-        points,
-        showStartPoint,
-        showEndPoint,
-        range,
-    } = props;
+    const {color, lineStyle, kind, points, showArrows, range} = props;
     const [point1, point2] = points;
+    // NOTE: Segments should not show both a point and an arrow.
+    //       Arrows supersede points.
+    const showStartPoint =
+        props.showStartPoint && (kind !== "segment" || !showArrows);
+    const showEndPoint =
+        props.showEndPoint && (kind !== "segment" || !showArrows);
+
+    let arrowHeadValues = getArrowheadValues(point1.coord, point2.coord, range);
+    let arrowTip = kind === "segment" ? point2.coord : arrowHeadValues.tip;
+    const startArrowHead = showArrows && (
+        <Arrowhead
+            angle={arrowHeadValues.angle}
+            tip={arrowTip}
+            color={lockedFigureColors[color]}
+        />
+    );
 
     let line;
 
     if (kind === "ray") {
-        // Rays extend to the end of the graph in one direction.
-        const endExtend = getIntersectionOfRayWithBox(
-            point2.coord,
-            point1.coord,
-            range,
-        );
         line = (
-            <Vector
-                tail={point1.coord}
-                tip={endExtend}
-                color={lockedFigureColors[color]}
-                style={{
-                    strokeDasharray:
-                        lineStyle === "dashed"
-                            ? // TODO(lems-1930): Uncomment this line when the
-                              // dashed style is updated in Mafs.
-                              // ? "var(--mafs-line-stroke-dash-style)"
-                              "4, 3"
-                            : undefined,
-                }}
-            />
+            <>
+                {showArrows && startArrowHead}
```

Comment:
`startArrowhead` is only set if `showArrows` is true, so it feels like one of the checks for `showArrows` could be removed.

Reply:
Yep. That was my intent, and then I just spaced it. Good catch!